### ISSUE TITLE
Support unboxed newtypes for primitives

### ISSUE THREAD

**Issue by joroKr21** [estatico/scala-newtype#13]
This is only possible with `NewSubType` but I think it's still useful.

**Comment by kailuowang** [comment]
Here is another motivation for adding NewSubType support to macro:I found that NewSubType works better with shapeless. (i.e. when deriving things using shapeless, case classes with field of NewType type has some kinkiness, while NewSubType seems just working fine. )

**Comment by carymrobbins** [comment]
A couple of ideas for how this would look -

```scala
@newtype(subtype = true) class Foo(x: Int)

@newtype.sub class Foo(x: Int)

@newsubtype class Foo(x: Int)
```

I'm not particularly happy with any of them, so if anyone has a better idea for the syntax, feel free to suggest it.

**Comment by carymrobbins** [comment]
This has been merged to master as `@newsubtype`, which will be available in the 0.4.0 release.
